### PR TITLE
Support Mobile Identity Connect

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
This PR adds support for MIC to the SDK.

#### Changes
- Override User class to handle redirect on iOS
- Utilize [nativescript-advanced-webview](https://github.com/bradmartin/nativescript-advanced-webview) to show web view.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,75 +11,43 @@
...
-    "es6-promise": "4.1.0",
-    "kinvey-html5-sdk": "3.5.0",
-    "kinvey-js-sdk": "file:///Users/Thomas/Documents/Kinvey/Development/SDKs/JavaScript/Core/SDK",
-    "lodash": "4.17.4"
-  },
-  "peerDependencies": {
-    "kinvey-html5-sdk": "3.5.0",
-    "kinvey-js-sdk": "3.5.0"
+    "events": "1.1.1",
+    "kinvey-js-sdk": "file:///Users/Thomas/Documents/Kinvey/Development/SDKs/JavaScript/Core/SDK"
```
Comment (tejasranade) [inline package.json]: @thomasconner needs to point to npm module `js-sdk`

File: src/middleware/src/cache.js [hunk 2]
```diff
@@ -1,8 +1,36 @@
-import { CacheMiddleware } from 'kinvey-js-sdk/dist/export';
-import Storage from './storage';
-
-export default class NativeScriptCacheMiddleware extends CacheMiddleware {
-  loadStorage(name) {
-    return new Storage(name);
-  }
-}
+"use strict";
```
Comment (tejasranade) [inline src/middleware/src/cache.js:1]: @thomasconner just a reminder to clean these compiled versions from the repo :)

File: src/middleware/src/http.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { device } from 'platform'
+
+function deviceInformation() {
+  const platform = device.os;
+  const version = device.osVersion;
+  const manufacturer = device.manufacturer;
+
+  // Return the device information string.
+  // const parts = [`js-${pkg.name}/${pkg.version}`];
+  const parts = ['js-kinvey-nativescript-sdk/3.5.0'];
```
Comment (tejasranade) [inline src/middleware/src/http.ts:12]: @thomasconner this needs to use the package name and number dynamically.

File: src/middleware/src/http.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    super(name);
+  }
+
+  handle(request: any) {
+    const { url, method, headers, body, timeout, followRedirect } = request;
+
+    // Add the X-Kinvey-Device-Information header
+    headers['X-Kinvey-Device-Information'] = deviceInformation();
+
+    return HttpRequest({
```
Comment (tejasranade) [inline src/middleware/src/http.ts:34]: @thomasconner nothing wrong with this, but I'm curious why xhr didn't work. Can you try to find out? Afaik, xhr should work just fine.

**Review by tejasranade — CHANGES_REQUESTED** [review]

File: src/middleware/src/http.ts (hunk 3 above)
Comment (thomasconner) [reply src/middleware/src/http.ts:12]: This is fixed in #2

File: src/middleware/src/cache.js (hunk 2 above)
Comment (thomasconner) [reply src/middleware/src/cache.js:1]: This is fixed in #2

File: package.json (hunk 1 above)
Comment (thomasconner) [reply package.json]: Fixed

**Pull request merged**